server/src/main/java/io/crate/action/sql/Session.java

```diff
@@ -697,22 +698,47 @@ private CompletableFuture<?> bulkExec(List<DeferredExecution> toExec) {
     private static void emitRowCountsToResultReceivers(UUID jobId,
                                                        JobsLogs jobsLogs,
                                                        List<DeferredExecution> executions,
-                                                       List<CompletableFuture<Long>> completedRowCounts) {
+                                                       List<CompletableFuture<Long>> completedRowCounts,
+                                                       boolean insertFailFast) {
+
         Object[] cells = new Object[1];
         RowN row = new RowN(cells);
+        Throwable errorEncountered = null;
         for (int i = 0; i < completedRowCounts.size(); i++) {
             CompletableFuture<Long> completedRowCount = completedRowCounts.get(i);
             ResultReceiver<?> resultReceiver = executions.get(i).resultReceiver();
+
+            // One error is encountered, finish all following results receivers
+            if (insertFailFast && errorEncountered != null) {
+                resultReceiver.allFinished();
+                continue;
+            }
+
```

Comment:
Not sure about this - I think we do need to finish remaining receivers but maybe need to use `resultReceiver.fail(errorEncountered)`?

I thought `fail` would be inappropriate as it could be different statements 
(for example `testExecuteBatchWithDifferentStatements), so having the same error can be really confusing?`